### PR TITLE
Fixed Filters, products pagination and sort Fix/608

### PR DESCRIPTION
**Pull request by AndronShermeta**
Fixed Filters, products pagination and sort

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
image |  image

image |
image

image |
image

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Comment by yuliiakucher** [comment]
image
please highlight the selected button

File: src/pages/product-list-page/product-sort/product-sort.js [hunk 1]
```diff
@@ -27,19 +27,31 @@ const ProductSort = () => {
...
+        case(Sortname.rate):{
+          dispatch(setSortByRate(optionValue.value));
+          break
+        }
+        case(Sortname.popularity):{
+          dispatch(setSortByPopularity(optionValue.value));
+          break
+        }
+        case Sortname.sortAsc:{
+          dispatch(setSortByPrice(optionValue.value+2));
```
Comment (yuliiakucher) [inline src/pages/product-list-page/product-sort/product-sort.js:52]: please change the value in const from -1 to 1

File: src/pages/product-list-page/product-list-filter/price-filter/price-filter.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ const PriceFilter = () => {
         onChangeCommitted={handlePriceFilter}
         valueLabelDisplay='auto'
         min={min}
-        max={max}
+        max={max+1}
```
Comment (yuliiakucher) [inline src/pages/product-list-page/product-list-filter/price-filter/price-filter.js:73]: please remove +1 a

File: src/redux/products/products.reducer.js [hunk 3]
```diff
@@ -106,7 +106,7 @@ const productsReducer = (state = initialState, action = {}) => {
     case SET_CURRENT_PAGE:
       return {
         ...state,
-        currentPage: action.payload
+        currentPage: action.payload - 1
```
Comment (grebiniuk) [inline src/redux/products/products.reducer.js:109]: Please fix in the other way

**Review by grebiniuk — CHANGES_REQUESTED** [review]

File: src/pages/product-list-page/count-per-page/count-per-page.js [hunk 4]
```diff
@@ -24,10 +27,10 @@ const CountPerPage = () => {
     searchParams.set(countPerPage, value);
     history.push(`?${searchParams.toString()}`);
   };
-
+  console.log(searchParams.get("countPerPage"))
```
Comment (Elugormo) [inline src/pages/product-list-page/count-per-page/count-per-page.js:30]: remove console.log

File: src/pages/product-list-page/product-list-filter/price-filter/price-filter.js [hunk 5]
```diff
@@ -54,8 +54,8 @@ const PriceFilter = () => {
     searchParams.set(page, defaultPage);
     history.push(`?${searchParams.toString()}`);
   };
-  const min = minPrice ? minPrice[currency].value / 100 : 0;
-  const max = maxPrice ? maxPrice[currency].value / 100 : 1000;
+  const min = parseInt(minPrice ? minPrice[currency].value / 100 : 0, 10);
+  const max = parseInt(maxPrice ? maxPrice[currency].value / 100 : 1000, 10);
```
Comment (Elugormo) [inline src/pages/product-list-page/product-list-filter/price-filter/price-filter.js:58]: Create utility function for these cases

File: src/redux/products/tests/products.reducer.test.js [hunk 6]
```diff
@@ -156,7 +156,7 @@ describe('Product reducer test', () => {
     };
     expect(productReducer(initialState, setPagesCount(10))).toEqual(state);
   });
-  it('should set current page to 4', () => {
+  it('should set current page to 3', () => {
```
Comment (Elugormo) [inline src/redux/products/tests/products.reducer.test.js:159]: fix "it" description

**Review by Elugormo — CHANGES_REQUESTED** [review]

File: src/utils/priceCalculating.js [hunk 7]
```diff
@@ -1 +1,3 @@
 export const calcPriceForCart = (item, currency) => item.price[currency].value * item.quantity;
+export const min = (minPrice,currency) => parseInt(minPrice ? minPrice[currency].value / 100 : 0, 10);
```
Comment (grebiniuk) [inline src/utils/priceCalculating.js:2]: You should make a new fucntion, like:

const roundPrice = (price) => parseInt(price/100, 10)
And use it in min/max functions and not duplicate the code.

**Review by grebiniuk — CHANGES_REQUESTED** [review]

File: src/utils/priceCalculating.js [hunk 8]
```diff
@@ -1 +1,4 @@
 export const calcPriceForCart = (item, currency) => item.price[currency].value * item.quantity;
+const roundPrice = (price) => parseInt(price/100, 10)
```
Comment (grebiniuk) [inline src/utils/priceCalculating.js:2]: Where is the semicolon?

**Review by grebiniuk — CHANGES_REQUESTED** [review]

**Review by grebiniuk — CHANGES_REQUESTED** [review]
Please fix conflicts

**Review by grebiniuk — APPROVED** [review]

**Review by yuliiakucher — APPROVED** [review]

**Review by Elugormo — APPROVED** [review]

**Pull request merged**